include/graphblas/bsp/collectives_blas1.hpp

```diff
@@ -392,105 +406,115 @@ namespace grb {
 		>
 		RC scatter(
```

Comment:
I have the feeling the de-duplication argument applies also to this primitives and to further ones.

Reply:
they are also scalar vs. array indeed

Reply:
(note btw as discussed a while ago-- the scalar variants do not register since we make the assumption it's small enough to make acceptable the overhead of copying into the local buffer, which is already registered-- thus saving a registration. A smarter implementation would take the ratio of local memory copy speed vs. network copy speed (plus network latency) to switch between the two variants, for both scalar and array cases, since scalars may be big enough / network fast enough to warrant registration while arrays may be small enough / network slow enough to warrant copying it into the buffer. Again though, the LPF collectives layer will have all the required information to perform such switching, and therefore should be done there-- except for one caveat: presently, `lpf_probe` cannot inform programs of local memory speed-- made LPF GitHub issue #20 for it)